extensions/PhabBugz/lib/Policy.pm

```diff
@@ -75,32 +79,17 @@ my $SearchResult = Dict[
 #   }
 # }
 
-#########################
-#    Initialization     #
-#########################
-
-sub new {
+sub new_from_query {
     my ($class, $params) = @_;
-    my $self = $params ? _load($params) : {};
-    $SearchResult->assert_valid($self);
-    return bless($self, $class);
-}
-
-sub _load {
-    my ($params) = @_;
     my $result = request('policy.query', $params);
     if (exists $result->{result}{data} && @{ $result->{result}{data} }) {
         return $result->{result}->{data}->[0];
     }
-    return $result;
+    return $class->new($result);
 }
```

Comment:
Ok this works fine after the below change. We were returning too early in new_from_query().

sub new_from_query {
    my ($class, $params) = @_;
    my $result = request('policy.query', $params);
    if (exists $result->{result}{data} && @{ $result->{result}{data} }) {
        return $class->new($result->{result}->{data}->[0]);
    }
    return undef;
}